packages/model-generator/src/create_graphql_document_generator.ts

```diff
@@ -38,10 +41,42 @@ export const createGraphqlDocumentGenerator = ({
   const fetchSchema = async () => {
     const backendOutputClient =
       BackendOutputClientFactory.getInstance(awsClientProvider);
-    const output = await getBackendOutputWithErrorHandling(
-      backendOutputClient,
-      backendIdentifier
-    );
+    let output;
+    try {
+      output = await getBackendOutputWithErrorHandling(
+        backendOutputClient,
+        backendIdentifier
+      );
+    } catch (error) {
+      if (
+        error instanceof BackendOutputClientError &&
+        error.code === BackendOutputClientErrorType.DEPLOYMENT_IN_PROGRESS
+      ) {
```

Comment:
This error handling should be happening in `getBackendOutputWithErrorHandling` which is its sole purpose.

Reply:
When debugging, I was seeing yargs will exit with the error message [here](https://github.com/aws-amplify/amplify-backend/pull/1790/files#diff-8cbab95ba8f311985c905dbe6e63fbaa448a8cec22c852b662760cddaba4293dR63) in the nested try catch but won't go up to the parent try catch in `getBackendOutputWithErrorHandling`. So we end up displaying the `BackendOutputClientError` and not the `AmplifyUserError` with resolution.

Reply:
How is that possible that the try catch in the `getBackendOutputWithErrorHandling` got skipped?

Reply:
That is what I was seeing and I didn't get the correct error until I wrapped `getBackendOutputWithErrorHandling` in a try catch. I was wondering if you've seen anything like this but I'll do more investigation.

Reply:
So I believe this had to do with `getBackendOutputWithErrorHandling` returning the async `getOutput`. Turning `getBackendOutputWithErrorHandling` async and awaiting on `getOutput` fixes this and I'm getting the correct error with resolution without the additional try catch here.